마이페이지 (내 정보) 구현

seongjin2427 opened the pull request:
# Pull Request

- [x] Feature

## Summary

My Page 중 내 정보 페이지 구현

## Describe your changes

- My Page 내 사이드 메뉴 구현 (추후 Admin 페이지와 공용으로 사용)
- profileImage, nickname, email, username, phoneNumber, gender, address, introduction, characteristic, address 정보 포함
- username, email 은 수정 불가능, read-only
- profileImage, nickname, introduction, characteristic은 수정하도록 구현
- react-hook-form 사용

## Issue number and link

#32 #34

seongjin2427 committed:
feat: MyPage Layout 및 SideMenu 구현

- MyPageLayout 1차 구현
- shared SideMenu 컴포넌트 구현
- MyPageMenu constants 선언

seongjin2427 committed:
feat: My page - 내 정보 페이지 1차 구현

- My page중 내 정보 페이지 1차 구현 (서버 전송은 미 구현)
- 프로필 사진 변경 기능
- 닉네임 변경 기능
- 상위 컴포넌트(profile page)에서 프롭스로 관련 데이터 받아 뿌리는 형식으로 구현

- 프로필 수정에 관한 프로세스 변경의 필요성을 느껴 임시저장을 위한
  커밋입니다.

seongjin2427 committed:
feat: Mypage - Profile 컴포넌트 분리 구현

- ProfileImage 컴포넌트 분리
- ProfileInfo 컴포넌트 분리
- profileImage, nickName, introduction, characteristic react-hook-form
  으로 구현

- 기본틀은 구현된 것 같고, 로그인 및 회원가입 구현 후, 아이디
  생성 및 로그인 하여 구현 예정

seongjin2427 committed:
feat: 소개글 및 성향 태그 갯수 제한 추가 및 소개글 높이 자동 증가

- 소개글 작성하는 만큼 높이 증가
- 태그 10개로 제한, 소개글 500자 제한
- 수정모드로 전환 시, 남은 숫자 / 전체 숫자 표기
- react-hook-form useFieldArray 객체 배열 -> 배열 변경 프로세스 추가
- opacity 0 -> 1 애니메이션 추가 (opacityIncrease)

seongjin2427 committed:
refact: My Page 내가 쓴 글, 내가 단 댓글, 좋아요한 글 페이지로 수정

- a, b, c 페이지 수정
- posts, replies, likes로 수정

seongjin2427 committed:
Merge branch 'develop' into feature/my-page

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
refact: temporary commit

jma1020 commented on constants/menus.ts:
```diff
@@ -1,3 +1,6 @@
+import { useRouter } from 'next/router';
```
여기 useRouter는 어디서 사용하나요??

seongjin2427 replied:
앗 현재는 삭제된 코드 입니다 ㅎㅎ

jma1020 commented on layouts/MyPageLayout/MyPageLayout.tsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <Styled.Container>
+      <Styled.SideMenu>
+        <SideMenu title="마이페이지" menus={menus} />
```
이름이 헷갈려서 Styled.SideMenuContainer나 Box로 바꾸는건 어떨까요?!

seongjin2427 replied:
피드백 감사합니다!
반영했어요!!

jma1020 commented on components/shared/ProfileImage/ProfileImage.tsx:
```diff
@@ -0,0 +1,101 @@
+  const imgRef = useRef<HTMLInputElement | null>(null);
+
+  if (register) {
+    var { ref, onChange, ...rest } = register('picture');
```
여기 var 키워드로 사용하셨는데 !! 이유가 있을까요?👀

seongjin2427 replied:
스코프 문제 때문에 잠시 사용했다가
지금은 리팩토링 하면서 변경 되었습니다 ㅎㅎ

seongjin2427 committed:
feat: my profile 추가 정보 포함하여 수정

- 프로필 정보 API 연동 (임시로 userId 4에 대한 정보를 가져옴)
- 프로필 정보에 대한 react-hook-form 스키마 (PROFILE_PATCH_SCHEMA) 작성
- 전체적인 코드 수정

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
feat: Profile Image 삭제 기능 추가

- profile image 삭제 기능 추가

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
fix: 삭제된 join 페이지 다시 포함

seongjin2427 committed:
temporary save

seongjin2427 committed:
merge

seongjin2427 committed:
Merge branch 'develop' into feature/my-page

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
refact: ProfileInfoTags 컴포넌트 분리

- ProfileInfo 컴포넌트 내 기능들이 많아 Tag 컴포넌트 영역을 ProfileInfoTags로 분리

seongjin2427 committed:
feat: Select 컴포넌트 pull 및 Profile 버전으로 추가

- ProfileInfoBirthDateSelect 컴포넌트 추가
- ProfileInfoRegionSelect 컴포넌트 추가
- Select 컴포넌트 props - type, isEdit 추가

seongjin2427 committed:
Merge branch 'feature/my-page' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
feat: 프로필 수정 기능 구현, 버그 존재하여 수정 요청

seongjin2427 committed:
feat: 프로필 이미지 업로드 기능 구현

seongjin2427 committed:
feat: user profile PATCH api 연동

- PATCH API 연동
- 한번 profile image 등록 후, 응답받은 base64 인코딩 문자열을 다시
  보내면 에러 발생하여 백엔드 담당자에게 문의 진행 중

seongjin2427 committed:
refact: 컴포넌트의 styled-components명 변경

- 컴포넌트 최상단 styled-component명 (Container -> 컴포넌트명+Container)

seongjin2427 committed:
Merge branch 'develop' into feature/my-page

seongjin2427 committed:
feat: SideMenu 컴포넌트 Backdrop 추가

seongjin2427 committed:
Merge branch 'feature/my-page' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
feat: Profile Image base64 -> File 변환 기능 추가

jma1020 commented on api/myPage.ts:
```diff
@@ -0,0 +1,147 @@
+import { base64ToFile } from '@/utils/image';
+import instance from './instance';
+
+interface TagsData {
```
혹시 타입들 파일 따로 분리하는건 어떨까요?!

seongjin2427 replied:
의견 감사합니다!
반영했습니다 ㅎㅎ

jma1020 commented on components/my/profile/ProfileInfoTags/ProfileInfoTags.styled.ts:
```diff
@@ -0,0 +1,132 @@
+  animation: ${opacityIncrease} 0.5s ease-in-out;
+`;
+
+export const TagsDiv = styled.div`
```
저희 컨벤션 정했던 wrapper로 바꾸면 어떨까요?!

seongjin2427 replied:
의견 감사합니다!
반영했습니다 ㅎㅎ

seongjin2427 committed:
refact: 피드백 반영

seongjin2427 committed:
Merge branch 'develop' of https://github.com/TRIPONG-2022/frontend into feature/my-page

seongjin2427 committed:
feat: Select 컴포넌트 원복 및 Profile에 맞춰서 Wrapper로 스타일 수정

sangrimlee reviewed (approved):
고생하셨어요 👍🏼

jma1020 reviewed (approved):
고생하셨습니다!!! 😄
